crates/bevy_reflect/src/map.rs

```diff
@@ -4,23 +4,32 @@ use bevy_utils::HashMap;
 
 use crate::{serde::Serializable, Reflect, ReflectMut, ReflectRef};
 
-/// An ordered `ReflectValue->ReflectValue` mapping. `ReflectValue` `Keys` are assumed to return a
-/// non-`None` hash. Ideally the ordering is stable across runs, but this is not required.
-/// This corresponds to types like [`std::collections::HashMap`].
+/// An ordered `ReflectValue->ReflectValue` mapping.
```

Comment:
Perhaps we could have a comment here explaining that the underlying types don't have to be the same?